services/src/datasets/external/aruna/mod.rs

```diff
@@ -212,106 +217,299 @@ impl ArunaDataProvider {
 
     /// Retrieves information for the dataset with the given id.
     async fn get_dataset_info(&self, id: &DataId) -> Result<ArunaDatasetIds> {
-        self.get_collection_info(Self::dataset_aruna_id(id)?).await
+        self.get_aruna_dataset_ids(Self::dataset_aruna_id(id)?)
+            .await
     }
 
     /// Retrieves information for the dataset with the given id.
     async fn get_dataset_info_from_layer(&self, id: &LayerId) -> Result<ArunaDatasetIds> {
-        self.get_collection_info(id.0.clone()).await
+        self.get_aruna_dataset_ids(id.0.clone()).await
+    }
+
+    fn get_outgoing_internal_relation_ids(
+        relations: &Vec<ArunaRelationStruct>,
+        target_resource_variant: ResourceVariant,
+    ) -> Result<Vec<String>> {
+        let mut ids = vec![];
+        for relation in relations {
+            match relation
+                .relation
+                .as_ref()
+                .ok_or(ArunaProviderError::MissingRelation)?
+            {
+                ArunaRelationEnum::External(_) => {}
+                ArunaRelationEnum::Internal(x) => {
+                    if x.direction() == RelationDirection::Outbound
+                        && x.resource_variant() == target_resource_variant
+                    {
+                        ids.push(x.resource_id.clone());
+                    }
+                }
+            }
+        }
+
+        Ok(ids)
     }
 
-    /// Retrieves all ids in the aruna object storage for the given `collection_id`.
-    async fn get_collection_info(&self, collection_id: String) -> Result<ArunaDatasetIds> {
-        let mut object_group_stub = self.object_group_stub.clone();
+    fn has_filter_label(&self, key_values: &Vec<KeyValue>) -> bool {
+        let label = &self.label_filter;
 
-        let mut aruna_object_group_overview = object_group_stub
-            .get_object_groups(GetObjectGroupsRequest {
-                collection_id: collection_id.clone(),
-                page_request: None,
-                label_id_filter: self.label_filter.clone(),
-            })
+        if let Some(filter) = label {
+            for key_value in key_values {
+                if key_value.variant() == KeyValueVariant::Label
+                    && key_value.key == *filter
```

Comment:
so key = value?

Reply:
Yes, that is the way we previously handled the labels in Aruna.